plugins/nodes/src/js/columns/NodesTableHostnameColumn.tsx

```diff
@@ -5,16 +5,39 @@ import Node from "#SRC/js/structs/Node";
 // import { IWidthArgs as WidthArgs } from "@dcos/ui-kit/packages/table/components/Column";
 import { IWidthArgs as WidthArgs } from "#PLUGINS/nodes/src/js/types/IWidthArgs";
 import { SortDirection } from "plugins/nodes/src/js/types/SortDirection";
+import { Link } from "react-router";
+import { Tooltip } from "reactjs-components";
+import Icon from "#SRC/js/components/Icon";
 
 export function hostnameRenderer(data: Node): React.ReactNode {
-  // TODO: DCOS-38819
-  return <span>{data.getHostName().toString()}</span>;
+  let headline = data.get("hostname");
```

Comment:
I find a little confusing to follow the `headline` variable, especially the `headline = "... {headline}..."` bit.

what do you think of separating the warning tooltip logic away from the cell?
```js
interface NodeProps {
  node: Node
}

const HostName:React.StatelessComponent<NodeProps> = ({node}) => {
  const hostname = node.get("hostname");
  
  if (node.isActive()) {
    return <span title={hostname}>{hostname}</span>;
  }

  const hostnameWithTooltip = (
    <Tooltip anchor="start" content="Connection to node lost">
      <Icon
        className="icon-alert icon-margin-right"
        color="neutral"
        id="yield"
        size="mini"
      />
      {hostname}
    </Tooltip>
  );

  return <span title={hostname}>{hostnameWithTooltip}</span>;
};

export function hostnameRenderer(data: Node): React.ReactNode {
  const nodeID = data.get("id");

  return (
    <Link className="table-cell-link-primary" to={`/nodes/${nodeID}`}>
      <HostName node={data} />
    </Link>
  );
}
```

Reply:
This code is going to be replaced in DCOS-39159, I'd vote not to invest more time than necessary here.

Also I dont agree with your solution Fabs, if `cellRenderer` gets too big, there is an issue with the abstraction and it shouldnt be fixed in that particular renderer - also I agree with you, that this renderer is too big 😅

IMHO the proper solution would be to split up the column into two (hostname & warningIcon).

Reply:
can we then just avoid?
```js
let headline = ...
headline = "... {headline}..."
```

Reply:
> This code is going to be replaced in DCOS-39159, I'd vote not to invest more time than necessary here.

You block it, Georgi changes it 🤷‍♂️

Reply:
I would like to see the variable naming fixed. Especially because we introduced now that we do:

```js
let headline = data.get("hostname");
headline = "... {headline}..."
...
...
<span title={data.get("hostname")}>{headline}</span>
```

Don't you think that if we had those names right on the beginning, we could have avoided the "object object" bug? 

(I am dismissing my block on any case, just would like that point to be considered)

Reply:
I agree, yes, but this is legacy code Georgi just copy & pasted… 🤷‍♂️

Reply:
@juliangieseke DCOS-39159 references the health/status column. Do you mean DCOS-39160? If so, this ticket is meant to clean up/refactor all the columns that were added? Shouldn't we just do it right in this ticket?

Reply:
As we decided that we need to own our copy and pasted legacy code I think we need to make this better. Let's get rid of the headline variable :)

Reply:
@DanielMSchmidt I have updated the code, please share your opinion.